scripts/format_code.sh

```diff
@@ -1,17 +1,12 @@
 #!/usr/bin/bash
 
-# Print usage() if no argument given
-if [ $# -eq 0 ]
-then
-    echo "./format_code.sh MY_DIR_TO_FORMAT"
-    exit 1
-fi
-
-# Retrieve directory to parse
-DIR=$1
+# Compute 'src' path
+SCRIPT_DIR=$(dirname "$0")
+REPO_DIR=$(readlink -f "$SCRIPT_DIR"/..)
+DIRS="$REPO_DIR/src $REPO_DIR/include"
 
 # Apply formating to all *.cpp and *.h files
-for FILE in `find "$DIR" -name '*.h' -o -name '*.cpp'`
+for FILE in $(find $DIRS -name '*.h' -o -name '*.cpp')
```

Comment:
Such declaration and usage of `DIRS` is vulnerable to whitespace or some special symbols in being present `"$REPO_DIR"`. The same problem applies to `for FILE in $(find ...)` (because of unquoted `$(...)`). The simplest solution is to `cd` to `"$REPO_DIR"`:

```bash
cd "$REPO_DIR"
DIRS="src include"

# Apply formating to all *.cpp and *.h files
for FILE in $(find $DIRS -name '*.h' -o -name '*.cpp')
...
```

Reply:
yes, fixing